python/lsst/ctrl/pool/parallel.py

```diff
@@ -473,3 +477,68 @@ def parseAndRun(cls, *args, **kwargs):
         super(BatchPoolTask, cls).parseAndRun(*args, **kwargs)
         pool.exit()
 
+
+class BatchTaskRunner(TaskRunner):
+    """Run a Task individually on a list of inputs using the MPI process pool"""
+    def __init__(self, *args, **kwargs):
+        TaskRunner.__init__(self, *args, **kwargs)
+        if self.numProcesses > 1:
+            self.log.warn("Multiprocessing arguments (-j %d) ignored since using batch processing" %
+                          self.numProcesses)
+            self.numProcesses = 1
+
+    def run(self, parsedCmd):
+        """Run the task on all targets"""
+        resultList = None
+
+        import multiprocessing
+        self.prepareForMultiProcessing()
+        pool = Pool()
+
+        if self.precall(parsedCmd):
+            targetList = self.getTargetList(parsedCmd)
+            if len(targetList) > 0:
+                parsedCmd.log.info("Processing %d targets with a pool of %d processes..." %
+                                   (len(targetList), pool.size))
+                # Run the task using self.__call__
+                resultList = pool.map(self, targetList)
+            else:
+                log.warn("Not running the task because there is no data to process; "
+                    "you may preview data using \"--show data\"")
+                resultList = []
+
+        return resultList
+
+    @abortOnError
+    def __call__(self, cache, args):
+        """Run the Task on a single target
+
+        Strips out the process pool 'cache' argument.
+
+        Brings down the entire job if an exception is not caught (i.e., --doraise).
+        """
+        return TaskRunner.__call__(self, args)
+
+
+class BatchParallelTask(BatchCmdLineTask):
+    """Runs the BatchCmdLineTask in parallel
+
+    Use this subclass of BatchCmdLineTask if you don't need to use the Pool
+    directly, but just want to iterate over many objects (like a multi-node
+    version of the '-j' command-line argument).
+    """
+    RunnerClass = BatchTaskRunner
+
+    @classmethod
+    def _makeArgumentParser(cls, *args, **kwargs):
+        kwargs.pop("doBatch", False)
+        kwargs.pop("add_help", False)
+        return super(BatchCmdLineTask, cls)._makeArgumentParser(*args, **kwargs)
+
+    @classmethod
+    def parseAndRun(cls, *args, **kwargs):
+        """Parse an argument list and run the command"""
+        pool = startPool()  # Start the pool early, to minimise the amount of duplicated work
+        results = super(BatchParallelTask, cls).parseAndRun(*args, **kwargs)
```

Comment:
I believe the coding style guideline suggests not using super, but since all the subclasses use it maybe it's o.k. in this case.